utils/utils.go

```diff
@@ -146,3 +148,26 @@ func GetNamespaceOrDefault(defaultNamespace string) string {
 	}
 	return namespace
 }
+
+// IsControlPlaneService returns true if the id corresponds to a Dapr
+// control plane service.
+func IsControlPlaneService(id string, additionalCPServices []string) bool {
+	switch id {
+	case "dapr-operator",
+		"dapr-placement",
+		"dapr-injector",
+		"dapr-sentry":
+		return true
+	}
+	var prefixedCPServices []string
+	if additionalCPServices != nil {
+		prefixedCPServices = make([]string, len(additionalCPServices))
+		for i, service := range additionalCPServices {
+			prefixedCPServices[i] = controlPlanePodNamePrefix + service
+		}
+	}
+	if len(prefixedCPServices) != 0 {
+		return slices.Contains(prefixedCPServices, id)
+	}
+	return false
```

Comment:
Small improvement that avoids allocating a new slice
```suggestion

	if len(additionalCPServices) == 0 {
		return false
	}

	after, found := strings.CutPrefix(id, controlPlanePodNamePrefix)
	return found && slices.Contains(additionalCPServices, after)
```